lib/cylc/cycling/__init__.py

```diff
@@ -20,9 +20,7 @@
 
 from abc import ABCMeta, abstractmethod
 
-from cylc.exceptions import (
-    CyclerTypeError, PointParsingError, IntervalParsingError,
-    SequenceDegenerateError)
+from cylc.exceptions import CyclerTypeError
```

Comment:
Now, this is a very interesting one. I only noticed it because Travis-CI failed. These three exceptions are not used in this file... but actually another file is importing them `from cylc.cycling`, which it quite bad. @cylc/core static analyzers, IDE, etc, are not able to pick this up... but if we started marking what was imported in each file explicitly through `__all__`, I think it would appear as a warning in linters/IDE (will create an issue with low priority later)

https://github.com/cylc/cylc/blob/8bc7aa096d00f92850a6af7ea22d0932a4782a63/lib/cylc/cycling/iso8601.py#L28-L30